src/tests/math/test_linspace.f90

```diff
@@ -0,0 +1,245 @@
+program test_linspace
+    use stdlib_error, only: check
+    use stdlib_kinds, only: sp, dp, int16, int8
+    use stdlib_math, only: linspace 
+
+    implicit none
+
+    integer :: iunit
+    logical :: warn = .false.
+
+    ! Testing linspace.
+    !
+    ! For single and double precision, check if the beginning and end values are properly calculated
+    ! and make sure that the size of the result is as expected.
+    !
+    !
+
+
+    open(newunit=iunit, file="test_linspace_log.txt", status="unknown") ! Log the results of the functio
+
+    write(iunit,*) "Writing to unit #: ", iunit
+
+    call test_linspace_sp
+    call test_linspace_dp
+    call test_linspace_neg_index ! Make sure that when passed a negative index the result is an empty array
+    call test_linspace_cmplx
+    call test_linspace_cmplx_2
+    call test_linspace_cmplx_3
+    call test_linspace_cmplx_sp
+    call test_linspace_cmplx_sp_2
+    call test_linspace_int16
+    call test_linspace_int8
```

Comment:
These are not tests. They just write the result to the standard output.

Reply:
I agree that they are not fully qualified as tests. I was wondering how I could actually test these functions. Do you have any ideas?

And btw, these subroutines actually output to `test_linspace_log.txt`, following the other testing convention that generates the `first_log_file.txt`, `second_log_file.txt`, ... , etc. This allows me to manually observe that the outputted values are inline with what we expect. I **know** that this is not a robust way to check for any time other than this development period, but I 'm not how else I can test these functions. 

I can check that the beginning and end values are as we expect, and that the length is correct, but that's about it.

Alternatively, I could call a simple linspace of say length 5: `linspace(0, 4, 5)`, and check that the values are exactly equal to 0, 1, 2, 3, and 4. Maybe I'll do that.

Reply:
> Alternatively, I could call a simple linspace of say length 5: `linspace(0, 4, 5)`, and check that the values are exactly equal to 0, 1, 2, 3, and 4. Maybe I'll do that.

I would suggest to do something like that.
An alternative is to check that the `array(i+1)-array(i)` is equal to the expcted value, for all `i`, in addition to check the first and last element of `array`.

Reply:
As I mentioned in my other reply to you, I am now testing that `abs( array(i + 1) - array(i) ) < `expected_value` * `TOLERANCE`), where `TOLERANCE = 0.0001`.